java/code/src/com/suse/manager/maintenance/MaintenanceManager.java

```diff
@@ -695,6 +696,57 @@ public boolean isActionInMaintenanceWindow(Action action, MaintenanceSchedule sc
         return false;
     }
 
+    private Collection<CalendarComponent> getDateFromSchedule(MaintenanceSchedule ms) {
+        return getScheduleEventsAtDate(new Date(), ms, parseCalendar(ms.getCalendarOpt().get()));
+    }
+
+    /**
+     * Check if system is in maintenance mode
+     *
+     * @param server the server to check
+     * @return true when the action is inside of a maintenance window, otherwise false
+     */
+    public static boolean checkIfInMaintenanceMode(MinionServer server) {
+        if (server.getMaintenanceScheduleOpt().isEmpty()) {
+            return true;
+        }
+        else {
+            MaintenanceManager mm = MaintenanceManager.instance();
+            Collection<CalendarComponent> cc = mm.getDateFromSchedule(server.getMaintenanceScheduleOpt().get());
+            return cc.isEmpty();
```

Comment:
I think we need to return `!cc.isEmpty()`.
CalendarComponents are mtching maintenance windows. So when we find something we are in maintenance mode,
otherwise - when empty - we are not.